### PR TITLE
feat(enhancment): More Motion Blur While Hyperspacing

### PR DESCRIPTION
**Pull request by Koranir**
## Feature Details
Makes the motion blur more intense while hyperspacing.

## UI Screenshots
[Screencast from 2023-07-07 13-49-13 (trimmed).webm](https://github.com/endless-sky/endless-sky/assets/101683475/4f599d42-0a01-40ef-94dc-1dfbbf48953b)

## Performance Impact
A little more is rendered while jumping, but no impact other than that.

### CODE REVIEW

**Comment by OcelotWalrus** [comment]
It's extremely cool I think but it would be better to add a setting like 'hyperspace blur' that you can set to 'off', 'low', 'medium', 'high'. And medium would be the current blur.

**Comment by Hurleveur** [comment]
I think we can already turn it on and off? Not sure but if so it'd make sense to add it to that, and I agree with OcelotWalrus

File: source/Ship.cpp [hunk 1]
```diff
@@ -2052,9 +2052,9 @@ bool Ship::IsEnteringHyperspace() const
 
 
 
-bool Ship::IsHyperspacing() const
+int Ship::IsHyperspacing() const
```
Comment (RisingLeaf) [inline source/Ship.cpp:2055]: should probablz be renamed to GetHyperspaceCount() as the is implies a bool return

**Review by RisingLeaf — CHANGES_REQUESTED** [review]

**Comment by Koranir** [comment]
Apparently, the starfield does not have motion blur disabled when motion blur should be disabled.

*why*

**Review by RisingLeaf — APPROVED** [review]

**Comment by Zitchas** [comment]
So long as this motion blur can be made to be disabled by the motion blur setting, then I'm ambivalent. Until then, however, I'm strongly against having more motion blur anywhere.

**Review by warp-core — COMMENTED** [review]
What's the reason for removing the existing method?

**Comment by Koranir** [comment]
> What's the reason for removing the existing method?

If you're talking about the IsHyperspacing() method, `int` implements the `operator bool` operator and has the exact same functionality if used with `logical and`s. I don't see a need to have two methaods that do the exact same things in the end.

**Comment by warp-core** [comment]
> > What's the reason for removing the existing method?
> 
> If you're talking about the IsHyperspacing() method, `int` implements the `operator bool` operator and has the exact same functionality if used with `logical and`s. I don't see a need to have two methaods that do the exact same things in the end.

I think calling a method named `IsHyperspacing` is a lot more meaningful when you want to know, well, if the ship is hyperspacing, than what's going on in this PR now. There's no reason to remove that. That method could even just call the new one and do the boolean conversion itself.

File: source/Engine.cpp [hunk 2]
```diff
@@ -501,6 +501,9 @@ void Engine::Step(bool isActive)
 	{
 		center = flagship->Position();
 		centerVelocity = flagship->Velocity();
+		// The starfield always has motion blur, so avoid increasing the blur if the player doesn't want it.
```
Comment (Amazinite) [inline source/Engine.cpp:504]: Isn't this comment no longer the case as of #9011?

**Comment by Koranir** [comment]
> > > What's the reason for removing the existing method?
> > 
> > 
> > If you're talking about the IsHyperspacing() method, `int` implements the `operator bool` operator and has the exact same functionality if used with `logical and`s. I don't see a need to have two methaods that do the exact same things in the end.
> 
> I think calling a method named `IsHyperspacing` is a lot more meaningful when you want to know, well, if the ship is hyperspacing, than what's going on in this PR now. There's no reason to remove that. That method could even just call the new one and do the boolean conversion itself.

Chaged to have two methods now.

**Comment by warp-core** [comment]
This also seems to add some motion blur to the starfield when using a jump drive, is that intentional?

**Comment by Koranir** [comment]
> This also seems to add some motion blur to the starfield when using a jump drive, is that intentional?

Yes

**Pull request merged**